alica_engine/src/engine/RuleBook.cpp

```diff
@@ -31,17 +31,19 @@ using std::endl;
 /**
  * Basic constructor
  */
-RuleBook::RuleBook(AlicaEngine* ae, PlanBase* pb)
-        : _tm(ae->getTeamManager())
-        , _ps(new PlanSelector(ae, pb))
-        , _log(ae->editLog())
+RuleBook::RuleBook(ConfigChangeListener& configChangeListener, Logger& log, SyncModule& synchModule, TeamObserver& teamObserver, TeamManager& teamManager,
+        const PlanRepository& planRepository, PlanBase* pb)
+        : _configChangeListener(configChangeListener)
```

Comment:
Only used in the constructor, we dont have to store the ConfigChangeListener locally